### ISSUE TITLE
Search box filter not working in dashboards

### ISSUE THREAD

**Issue by clanog** [metabase/metabase#50404]
### Describe the bug

The search box filter when the filter operator is `Is` is not working.

We have this filter in a column with company names and we would like the filter to allow to search for any company. It doesn't. 

What it looks like it's happening is that the optional filter values are preloaded (as would be expected with a dropdown list, not search box) and when we want to search for a value outside of those, search simply doesn't work.

It works though if the literal value that the filter should returned is copied.

### To Reproduce

Create a dashboard with a filter.

The filter has to be:
- text
- search box
- filter rule: is

Preferably linked to a column with a lot of values (say thousands of values)

### Expected behavior

Search works

### Information about your Metabase installation

```JSON
- Can reproduce in both Firefox and Chrome

Metabase open source version v0.51.4
```

### Severity

blocking some users

**Comment by romeovs** [comment]
@clanog if I understand correctly, the issue is that it's not possible to select values outside of the initially loaded list of values? But the filter _does_ actually filter if the value to be selected is inside that list?

If so I this is a duplicate of https://github.com/metabase/metabase/issues/49642 (in which case it is will have been fixed by https://github.com/metabase/metabase/pull/49739).

If not, please share some more information about what exactly is not working.

**Comment by romeovs** [comment]
This was released in [v52.1](https://github.com/metabase/metabase/releases/tag/v0.52.1).

Please close the issue if this resolved your issue.
